### PR TITLE
Initial import

### PR DESCRIPTION
**Pull request by flyofsky**
Add default project appium-sdb.
appium-sdb is used by the appium-tizen-driver.

### CODE REVIEW

File: lib/tools/system-calls.js [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    } else {
+      log.error(`Could not find device.`);
+      return false;
+    }
+  } catch (e) {
+    log.errorAndThrow("Error while connect devices.");
+  }
+};
+
+systemCallMethods.getDevicesWithRetry = async function (timeoutMs = 20000) {
```
Comment (rookiejava) [inline lib/tools/system-calls.js]: why don't use  DEFAULT_SDB_EXEC_TIMEOUT?

File: lib/tools/system-calls.js [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+    if ((Date.now() - start) > timeoutMs) {
+      throw new Error("Could not find a connected tizen device.");
+    }
+    try {
+      let devices = await this.getConnectedDevices();
+      if (devices.length < 1) {
+        log.debug("Could not find devices, restarting sdb server...");
+        await this.restartSdb();
+
+        await sleep(1000);
```
Comment (rookiejava) [inline lib/tools/system-calls.js]: why sleep require here?

File: lib/tools/system-calls.js [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+
+systemCallMethods.restartSdb = async function () {
+  if (this.suppressKillServer) {
+    log.debug(`Not restarting sdb since 'suppressKillServer' is on`);
+    return;
+  }
+
+  log.debug('Restarting sdb');
+  try {
+    await exec(this.executable.path, ['kill-server']);
```
Comment (rookiejava) [inline lib/tools/system-calls.js:105]: why not start server here?

File: lib/tools/sdb-commands.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+let methods = {};
+
+methods.getSdbWithCorrectSdbPath = async function () {
+  this.executable.path = await this.getSdkBinaryPath("sdb");
+  this.binaries.sdb = this.executable.path;
+  return this.sdb;
+};
+
+methods.isDeviceConnected = async function () {
+  let devices = await this.getConnectedDevices();
```
Comment (rookiejava) [inline lib/tools/sdb-commands.js]: Use `sdb get-state`  to determine whethere device connected (`device`) or not (`offline` or `locked` or `unknown`).

File: lib/tools/system-calls.js [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+    cmd = "where";
+  }
+  return cmd;
+};
+
+systemCallMethods.setEmulatorPort = function (emPort) {
+  this.emulatorPort = emPort;
+};
+
+systemCallMethods.installRpm = async function (rpm, path, opts = {}) {
```
Comment (rookiejava) [inline lib/tools/system-calls.js]: do we need to provide install/uninstall rpm ?

**Review by rookiejava — CHANGES_REQUESTED** [review]

File: lib/tools/system-calls.js (hunk 1 above)
Comment (flyofsky) [reply lib/tools/system-calls.js]: Done

File: lib/tools/system-calls.js (hunk 2 above)
Comment (flyofsky) [reply lib/tools/system-calls.js]: Done

File: lib/tools/sdb-commands.js (hunk 4 above)
Comment (flyofsky) [reply lib/tools/sdb-commands.js]: Done

File: lib/tools/system-calls.js (hunk 3 above)
Comment (flyofsky) [reply lib/tools/system-calls.js:105]: Done

File: lib/tools/system-calls.js (hunk 5 above)
Comment (flyofsky) [reply lib/tools/system-calls.js]: Done

File: lib/tools/system-calls.js [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+import { system } from 'appium-support';
+import { exec } from 'teen_process';
+import { retry } from 'asyncbox';
+import _ from 'lodash';
+
+let systemCallMethods = {};
+
+const DEFAULT_SDB_EXEC_TIMEOUT = 20000;
+
+systemCallMethods.setWriteFileSystem = async function () {
```
Comment (rookiejava) [inline lib/tools/system-calls.js]: Is it required ?

Comment (flyofsky) [reply lib/tools/system-calls.js]: Done

File: README.md [hunk 7]
```diff
@@ -1,2 +1,61 @@
...
+- `setDeviceId`
+- `reboot`
+- `isAppInstalled`
+- `startApp`
+- `uninstall`
+- `installFromDevicePath`
+- `install`
+- `root`
+- `unroot`
+- `setWriteFileSystem`
```
Comment (rookiejava) [inline README.md]: need to update too

Comment (flyofsky) [reply README.md]: Done

**Review by rookiejava — APPROVED** [review]

**Review by rookiejava — COMMENTED** [review]
Should provide docs and test.

**Pull request merged**